4.5.1: reset_sequences=True fails on MariaDB/MySQL

Firstly, thanks for maintaining such a powerful and useful testing library for Django.

On to the bug:

- OS: Windows 10
- Python: 3.9.1
- pytest-6.2.5
- py-1.11.0
- pluggy-1.0.0
- Django: 3.2.10

Example:

    @pytest.mark.django_db(reset_sequences=True)
    def test_reset_sequences():
        assert True

Output:

    ERROR my_test.py::test_reset_sequences - AssertionError: reset_sequences cannot be used on TestCase instances

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -1,6 +1,16 @@
 Changelog
 =========
 
+v4.5.2 (2021-12-07)
+-------------------
+
+Bugfixes
+^^^^^^^^
+
+* Fix regression in v4.5.0 - ``pytest.mark.django_db(reset_sequence=True)`` now
+  implies ``transaction=True`` again.
+
+
 v4.5.1 (2021-12-02)
 -------------------
 

--- a/pytest_django/fixtures.py
+++ b/pytest_django/fixtures.py
@@ -167,7 +167,7 @@ def _django_db_helper(
             serialized_rollback,
         ) = False, False, None, False
 
-    transactional = transactional or (
+    transactional = transactional or reset_sequences or (
         "transactional_db" in request.fixturenames
         or "live_server" in request.fixturenames
     )

--- a/tests/test_database.py
+++ b/tests/test_database.py
@@ -287,8 +287,13 @@ class TestDatabaseMarker:
         marker = request.node.get_closest_marker("django_db")
         assert not marker.kwargs
 
-    @pytest.mark.django_db(transaction=True, reset_sequences=True)
+    @pytest.mark.django_db(reset_sequences=True)
     def test_reset_sequences_enabled(self, request) -> None:
+        marker = request.node.get_closest_marker("django_db")
+        assert marker.kwargs["reset_sequences"]
+
+    @pytest.mark.django_db(transaction=True, reset_sequences=True)
+    def test_transaction_reset_sequences_enabled(self, request) -> None:
         marker = request.node.get_closest_marker("django_db")
         assert marker.kwargs["reset_sequences"]
 

--- a/tox.ini
+++ b/tox.ini
@@ -12,7 +12,7 @@ envlist =
 extras = testing
 deps =
     djmain: https://github.com/django/django/archive/main.tar.gz
-    dj40: Django>=4.0a1,<4.1
+    dj40: Django>=4.0,<4.1
     dj32: Django>=3.2,<4.0
     dj31: Django>=3.1,<3.2
     dj22: Django>=2.2,<2.3

